src/openlcb/MemoryConfigClient.cxxtest

```diff
@@ -131,6 +188,255 @@ TEST_F(MemoryConfigClientTest, create)
+    //
+    expect_any_packet();
+    EXPECT_CALL(tSpace_, get_read_timeout())
+        .WillRepeatedly(Return(DatagramDefs::TIMEOUT_NONE));
+    EXPECT_CALL(tSpace_, read(0, _, 64, _, NotifyCapture(&done)))
+        .WillOnce(DoAll(
+            SetArgPointee<3>(0x3FFF),
+            Return(0)));
+    b = invoke_client_no_block(
+        MemoryConfigClientRequest::READ, NodeHandle(TEST_NODE_ID), 0x52);
+    wait();
+    clock.advance(MSEC_TO_NSEC(16200));
+    wait();
+    EXPECT_EQ(Defs::OPENMRN_TIMEOUT, b->data()->resultCode);
+    ASSERT_EQ(0U, b->data()->payload.size());
+    EXPECT_TRUE(clientTwo_.is_waiting());
+    testing::Mock::VerifyAndClearExpectations(&tSpace_);
+    expect_any_packet();
+    EXPECT_CALL(tSpace_, read(0, _, 64, _, _))
+        .WillOnce(DoAll(WithArgs<0, 1, 2>(
+                Invoke(this, &MemoryConfigClientTest::FillPayload)),
+            SetArgPointee<3>(MemoryConfigDefs::ERROR_OUT_OF_BOUNDS),
+            Return(24)));
+    done->notify();
+    wait();
+    ASSERT_EQ(0U, b->data()->payload.size());
+    testing::Mock::VerifyAndClearExpectations(&tSpace_);
+}
+
+// Tests using the MemoryConfigClient to write a payload that fits into one
+// datagram.
+TEST_F(MemoryConfigClientTest, writeTimeout)
+{
+    using testing::DoAll;
+    using testing::WithArgs;
+    using ::testing::SetArgPointee;
+
+    Notifiable *done;
+    FakeClock clock;
+
+    // Initialize test data.
+    string test_payload;
+    for (size_t i = 0; i < dataContents2_.size(); ++i)
+    {
+        test_payload.push_back(i + 56);
+    }
+    EXPECT_NE(0,
+        memcmp(&dataContents_[34], test_payload.data(), test_payload.size()));
+
+
+    //
+    // Pass back timeout "none". Don't actually wait to timeout.
+    //
+    expect_any_packet();
+    EXPECT_CALL(tSpace_, get_write_timeout())
+        .WillRepeatedly(Return(DatagramDefs::TIMEOUT_NONE));
+    EXPECT_CALL(tSpace_, write(0, _, test_payload.size(), _, _))
+        .WillOnce(DoAll(WithArgs<0, 1, 2>(
+                Invoke(this, &MemoryConfigClientTest::AcceptPayload)),
+            Return(test_payload.size())));
+    auto b = invoke_flow(&clientTwo_, MemoryConfigClientRequest::WRITE,
+        NodeHandle(TEST_NODE_ID), 0x52, 0, test_payload);
+    wait();
+    EXPECT_EQ(0, b->data()->resultCode);
+    ASSERT_EQ(dataContents2_.size(), b->data()->payload.size());
+    EXPECT_FALSE(
+        memcmp(&dataContents2_[0], test_payload.data(), test_payload.size()));
+    testing::Mock::VerifyAndClearExpectations(&tSpace_);
+
+
+    //
+    // Pass back timeout "none". Force a timeout (3 seconds).
+    //
+    expect_any_packet();
+    init_data_contents2();
+    EXPECT_CALL(tSpace_, get_write_timeout())
+        .WillRepeatedly(Return(DatagramDefs::TIMEOUT_NONE));
+    EXPECT_CALL(
+        tSpace_, write(0, _, test_payload.size(), _, NotifyCapture(&done)))
+            .WillOnce(DoAll(
+                SetArgPointee<3>(0x3FFF), Return(test_payload.size())));
```

Comment:
I think this should be Return(0) since you are not actually accepting the payload

Reply:
I have revisited this. Both flows are possible, but the program track does actually use your flow. I updated the test to use your flow.